src/auto-line-wipe/MultiAutoLineWipeSpell.sol

```diff
@@ -0,0 +1,99 @@
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity ^0.8.16;
+
+import {DssEmergencySpell} from "../DssEmergencySpell.sol";
+
+interface IlkRegistryLike {
+    function count() external view returns (uint256);
+    function list() external view returns (bytes32[] memory);
+    function list(uint256 start, uint256 end) external view returns (bytes32[] memory);
+}
+
+interface LineMomLike {
+    function autoLine() external view returns (address);
+    function ilks(bytes32 ilk) external view returns (uint256);
+    function wipe(bytes32 ilk) external;
+}
+
+interface AutoLineLike {
+    function ilks(bytes32 ilk)
+        external
+        view
+        returns (uint256 maxLine, uint256 gap, uint48 ttl, uint48 last, uint48 lastInc);
+}
+
+contract MultiAutoLineWipeSpell is DssEmergencySpell {
+    string public constant override description = "Emergency Spell | Multi AutoLine Wipe";
+
+    IlkRegistryLike public immutable ilkReg = IlkRegistryLike(_log.getAddress("ILK_REGISTRY"));
+    LineMomLike public immutable lineMom = LineMomLike(_log.getAddress("LINE_MOM"));
+
+    event Wipe(bytes32 indexed ilk);
+
+    /**
+     * @notice Wipes, when possible, all ilks from auto-line;
+     */
+    function _emeregencyActions() internal override {
+        bytes32[] memory ilks = ilkReg.list();
+        _doWipe(ilks);
+    }
+
+    /**
+     * @notice Wipe all ilks in the batch from auto-line.
+     * @dev This is an escape hatch to prevent this spell from being blocked in case it would hit the block gas limit.
+     *      In case `end` is greater than the ilk registry length, the iteration will be automatically capped.
+     * @param start The index to start the iteration (inclusive).
+     * @param end The index to stop the iteration (inclusive).
+     */
+    function stopBatch(uint256 start, uint256 end) external {
+        uint256 maxEnd = ilkReg.count() - 1;
+        bytes32[] memory ilks = ilkReg.list(start, end < maxEnd ? end : maxEnd);
+        _doWipe(ilks);
+    }
+
+    /**
+     * @notice Stops, when possible, all OSMs that can be found from the `ilks` list.
+     * @param ilks The list of ilks to consider.
+     */
+    function _doWipe(bytes32[] memory ilks) internal {
+        for (uint256 i = 0; i < ilks.length; i++) {
+            if (lineMom.ilks(ilks[i]) == 0) continue;
+
+            LineMomLike(lineMom).wipe(ilks[i]);
```

Comment:
Here if one of the wipe fails the whole spell will fail (this could happen due to a wrong auth/ward setup for example). In an emergency situation it could be better to wipe all that are possible, and emit an event for failures to be analyzed afterwards.

Reply:
Wipe can if, and only if:
- `ilk` was not added to `LineMom` ([L106](https://github.com/makerdao/line-mom/blob/5935e56ac8d4d677865259e94129bf2bd3039ebf/src/LineMom.sol#L106)): we already guard against that condition on L74.
- `LineMom` is not a ward on the `Vat` ([L108](https://github.com/makerdao/line-mom/blob/5935e56ac8d4d677865259e94129bf2bd31039ebf/src/LineMom.sol#L108)): in this case, `wipe` would already fail for all ilks.
- `LineMom` is not a ward on `AutoLine` ([L109](https://github.com/makerdao/line-mom/blob/5935e56ac8d4d677865259e94129bf2bd31039ebf/src/LineMom.sol#L109)): same as above, the call would revert.